tests/internal/crashtracker/test_crashtracker.py

```diff
@@ -0,0 +1,265 @@
+
+    data = utils.conn_to_bytes(conn)
+    conn.close()
+    assert b"string_at" in data
+
+
+@pytest.mark.skipif(not sys.platform.startswith("linux"), reason="Linux only")
+@pytest.mark.subprocess()
+def test_crashtracker_simple_sigbus():
+    # This is similar to the simple fork test, except instead of raising a SIGSEGV,
+    # it organically raises a real SIGBUS.
+    import ctypes
+    from ctypes.util import find_library
+    import os
+    import tempfile
+
+    import tests.internal.crashtracker.utils as utils
+
+    # Part 0, set up the interface to mmap.  We don't want to use mmap.mmap because it has too much protection.
+    libc = ctypes.CDLL(find_library("c"))
+    assert libc
+    libc.mmap.argtypes = [ctypes.c_void_p, ctypes.c_size_t, ctypes.c_int, ctypes.c_int, ctypes.c_int, ctypes.c_long]
+    libc.mmap.restype = ctypes.c_void_p
+    PROT_WRITE = ctypes.c_int(0x2)  # Maybe there's a better way to get this constant?
+    MAP_PRIVATE = ctypes.c_int(0x02)
+
+    # Part 1 and 2
+    port, sock = utils.crashtracker_receiver_bind()
+    assert port
+    assert sock
+
+    # Part 3, setup crashtracker in parent
+    assert utils.start_crashtracker(port)
+    stdout_msg, stderr_msg = utils.read_files(["stdout.log", "stderr.log"])
+    assert not stdout_msg
+    assert not stderr_msg
+
+    # Part 4, Fork and crash
+    pid = os.fork()
+    if pid == 0:
+        with tempfile.TemporaryFile() as tmp_file:
+            tmp_file.write(b"aaaaaa")  # write some data to the file
+            fd = tmp_file.fileno()  # get the file descriptor
+            mm = libc.mmap(
+                ctypes.c_void_p(0), ctypes.c_size_t(4096), PROT_WRITE, MAP_PRIVATE, ctypes.c_int(fd), ctypes.c_long(0)
+            )
+            assert mm
+            assert mm != ctypes.c_void_p(-1).value
+            arr_type = ctypes.POINTER(ctypes.c_char * 4096)
+            arr = ctypes.cast(mm, arr_type).contents
+            arr[4095] = b"x"  # sigbus
+        exit(-1)  # just in case
+
+    # Part 5, check
+    conn = utils.listen_get_conn(sock)
+    assert conn
+
+    data = utils.conn_to_bytes(conn)
+    conn.close()
+    assert data
+
+
+@pytest.mark.skipif(not sys.platform.startswith("linux"), reason="Linux only")
+@pytest.mark.subprocess()
+def test_crashtracker_raise_sigsegv():
+    import os
+    import signal
+
+    import tests.internal.crashtracker.utils as utils
+
+    # Part 1 and 2
+    port, sock = utils.crashtracker_receiver_bind()
+    assert port
+    assert sock
+
+    assert utils.start_crashtracker(port)
+    stdout_msg, stderr_msg = utils.read_files(["stdout.log", "stderr.log"])
+    assert not stdout_msg
+    assert not stderr_msg
+
+    # Part 4, raise SIGSEGV
+    pid = os.fork()
+    if pid == 0:
+        os.kill(os.getpid(), signal.SIGSEGV.value)
+        exit(-1)
+
+    # Part 5, check
+    conn = utils.listen_get_conn(sock)
+    assert conn
+
+    data = utils.conn_to_bytes(conn)
+    conn.close()
+    assert b"os_kill_impl" in data
```

Comment:
Where does the _impl come from?

Reply:
It's an internal symbol in cpython.  I don't know precisely where it is defined, but I actually can't see it in the strings for cpython 3.12.  I'll drop the `_impl` part in this test.